CMakeLists.txt

```diff
@@ -29,18 +29,24 @@
 cmake_minimum_required(VERSION 3.13.1)
 project(ot-commissioner VERSION 1.0.0)
 
+option(OT_COMM_COVERAGE         "Enable coverage reporting" OFF)
+option(OT_COMM_TEST             "Build tests" ON)
+option(OT_COMM_APP              "Build the CLI App" ON)
+option(OT_COMM_MAINTAINER_FLAGS "Build flags used by the maintainers" ON)
+
 if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Debug)
 endif()
 
-option(OT_COMM_COVERAGE "Enable coverage reporting" OFF)
-option(OT_COMM_TEST     "Build tests" ON)
-option(OT_COMM_APP      "Build the CLI App" ON)
+if(NOT "${CMAKE_CXX_STANDARD}")
+    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -Werror -Wfatal-errors -Wno-missing-braces")
+if(OT_COMM_MAINTAINER_FLAGS)
```

Comment:
In which case will this option be useful? I think those FLAGS has effect on only OT Commissioner project and dependencies.

Reply:
E.g. using yocto: the conpiler flags are set globally for all libraries and executables... this was resetting those settings

Reply:
you meaning using yocto as a library of OT Commissioner?

Reply:
yocto is a build system to create a linux distribution. this system is widely used to create costum industrial grade linux-based OSs.

Yocto relies on recipes which use the native build system e.g. cmake to install artifacts to the rootfs of the OS image.

yocto will use different flags depending on the machine/architecture/rootfs etc.

those flags are passed to the recipes and build systems e.g. cmake, autotools, scons, meson etc.

Those flags have to be respected by the libraries in order to have a common ABI and link different libs and executables together...

Reply:
>Those flags have to be respected by the libraries in order to have a common ABI and link different libs and executables together...

it seems current `CMAKE_CXX_FLAGS` specifies only warning strategies. For your requirement of passing extra FLAGS to OT Commissioner, how about we set `CMAKE_CXX_FLAGS` as:
```cmake

```set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Wfatal-errors -Wno-missing-braces")

Reply:
top

Reply:
so, but this means no support for msvc?

Reply:
We current don't support Windows. May improve later to support Windows.